frontend/packages/dev-console/src/components/import/import-submit-utils.ts

```diff
@@ -235,5 +240,26 @@ export const createRoute = (
     },
   };
 
-  return k8sCreate(RouteModel, route);
+  return k8sCreate(RouteModel, route, dryRunOpt);
+};
+
+export const createResources = (
+  formData: GitImportFormData,
+  imageStream: K8sResourceKind,
+  dryRunOpt: { queryParams?: { [key: string]: string } } = {},
+): Promise<K8sResourceKind>[] => {
+  const {
+    route: { create: canCreateRoute },
+    image: { ports },
+  } = formData;
+
+  const requests: Promise<K8sResourceKind>[] = [
+    createDeploymentConfig(formData, imageStream, dryRunOpt),
+    createImageStream(formData, imageStream, dryRunOpt),
+    createBuildConfig(formData, imageStream, dryRunOpt),
+    ...(!_.isEmpty(ports) ? [createService(formData, imageStream, dryRunOpt)] : []),
+    ...(!_.isEmpty(ports) && canCreateRoute ? [createRoute(formData, imageStream, dryRunOpt)] : []),
```

Comment:
This method of adding to the array is trying to be a bit fancy and it would be more clear to use if conditions afterwards with array push.

Reply:
I will change it to if conditions.

Reply:
Done